### DISCUSSION TITLE
Particle Iteration description

### DISCUSSION THREAD

**Discussion by naidoo88** [dglazier/chanser#24] [User problems]
Hey Derek,

I've had a really good read of the [Particle Iterator Algorithm](https://github.com/dglazier/chanser/blob/master/ParticleIteratorAlgorithm.md), but I'm having some troubles following the description of what it happening.  I can follow the general gist of it, but when it reaches the line with letter/numbered list for the pi0 example it totally loses me.  I can't marry it together with the description above.  

It could well be my brain, but a proper visual illustration, even just a sort of 'boxes and arrows' flow diagram, would be amazing to help pull together the written description above it.  

As users are going to depend on this algorithm to some degree or another, I'm sure it would be useful to others down the line too.

Cheers,
Paul

**Comment by dglazier** [comment]
Noted. I thought I had written something much simpler and clearer on this but can't find it. In the meantime  if you like you can uncomment the CheckCombitorial function on FinalState.cpp and rebuild. It will then output all the combitorial events to screen for you to check and make sure it is behaving as you would like. I will add a flag to turn this off and on as you should not really need to edit the source code....

void FinalState::FSProcess(){
    
    _rejectEvent=kFALSE;//in principle will save this combi
    //CheckCombitorial();  <--- this line causes text outpur to screen showing the combitorials
